torch/distributed/fsdp/_state_dict_utils.py

```diff
@@ -586,10 +590,10 @@ def _sharded_pre_load_state_dict_hook(
             "are flattened and sharded."
         )
 
-    tensors_to_flatten: List[torch.Tensor] = []
-    shared_param_fqns = [
-        fqn for fqn, _, _ in _shared_param_name_infos(module, fsdp_state)
-    ]
+    fqn_to_param_ext = {
+        t[0]: t[1]
+        for t in zip(handle.flat_param._fqns, handle.flat_param._param_extensions)
```

Comment:
nit: maybe for readability?
```suggestion
        fqn: param_extension
        for fqn, param_extension in zip(handle.flat_param._fqns, handle.flat_param._param_extensions)
```